### PR TITLE
www: show global, friends and user statistics

### PR DESCRIPTION
**Pull request by willeponken**
Fixes: #172 
Fixes: #164 
Fixes: #161 
Fixes: #155

### CODE REVIEW

File: www/src/models/statistics.ts [hunk 1]
```diff
@@ -1,62 +1,249 @@
...
+import Network from '../services/network';
+
+const colors = {
+  red: 'rgb(250, 50, 47)',
+  orange: 'rgb(203, 75, 22)',
+  yellow: 'rgb(181, 137, 0)',
+  green: 'rgb(133, 153, 0)',
+  blue: 'rgb(38, 139, 210)',
+  magenta: 'rgb(211, 54, 130)',
+  grey: 'rgb(131, 148, 150)',
```
Comment (Hjortsberg) [inline www/src/models/statistics.ts]: i assume we're deliberately deviating from the solarized theme, or is this another one of the 10-ish grey defined colors in the theme?

File: www/src/models/statistics.ts [hunk 2]
```diff
@@ -1,62 +1,249 @@
...
+  violet: 'rgb(108, 113, 196)',
+  cyan: 'rgb(0, 181, 173)',
```
Comment (Hjortsberg) [inline www/src/models/statistics.ts]: Here too, assume that this color deviates due to UX reasons?

File: www/src/models/statistics.ts (hunk 1 above)
Comment (willeponken) [reply www/src/models/statistics.ts]: I think it's one of those other grey colors, but I took it directly from the Solar theme, it seems like it deviates from the original Solarized.

**Comment by willeponken** [comment]
Kindly review my latest work, Sir. @Hjortsberg

File: model/statistics.go [hunk 3]
```diff
@@ -151,23 +122,34 @@ func getLinesOfCode(username string) (lines LinesOfCode, err error) {
...
 		      LOWER(username)=LOWER(TRIM($1))
 	`, username).Scan(&numberOfRoasts.Count)
-	if err != nil {
-		return
-	}
+	return
+}
 
+// GetRoastRatio returns the lines of code, number of errors and warnings for
+// everyone or a specific user. An Empty string as usernmae represents everyone.
```
Comment (Hjortsberg) [inline model/statistics.go]: username*

Comment (willeponken) [reply model/statistics.go]: :deciduous_tree:

**Comment by Hjortsberg** [comment]
MERGE! GOAHEAD

**Pull request merged**